runtime/near-vm-runner/src/wasmer1_runner.rs

```diff
@@ -239,8 +238,77 @@ fn run_method(module: &Module, import: &ImportObject, method_name: &str) -> Resu
     Ok(())
 }
 
-pub fn compile_module(code: &[u8]) -> bool {
-    let engine = JIT::new(Singlepass::default()).engine();
-    let store = Store::new(&engine);
+pub(crate) fn compile_wasmer1_module(code: &[u8]) -> bool {
+    let store = default_wasmer1_store();
     Module::new(&store, code).is_ok()
 }
+
+pub(crate) fn default_wasmer1_store() -> Store {
+    let use_cranelift = false;
+    let use_native_engine = false;
+    if use_native_engine {
+        let engine = if use_cranelift {
+            wasmer_engine_native::Native::new(wasmer_compiler_cranelift::Cranelift::default())
+                .engine()
+        } else {
+            wasmer_engine_native::Native::new(wasmer_compiler_singlepass::Singlepass::default())
+                .engine()
+        };
+        Store::new(&engine)
+    } else {
+        let engine = JIT::new(Singlepass::default()).engine();
+        Store::new(&engine)
+    }
+}
+
+pub(crate) fn run_wasmer1_module<'a>(
+    module: &Module,
+    store: &Store,
+    method_name: &str,
+    ext: &mut dyn External,
+    context: VMContext,
+    wasm_config: &'a VMConfig,
+    fees_config: &'a RuntimeFeesConfig,
+    promise_results: &'a [PromiseResult],
+    profile: ProfileData,
+    current_protocol_version: ProtocolVersion,
+) -> (Option<VMOutcome>, Option<VMError>) {
+    // Do we really need that code?
+    if method_name.is_empty() {
+        return (
+            None,
+            Some(VMError::FunctionCallError(FunctionCallError::MethodResolveError(
+                MethodResolveError::MethodEmptyName,
+            ))),
+        );
+    }
```

Comment:
:+1: at minimum, we need a

```
struct MethodName<'a>(&'a str);

impl<'a> TryFrom<&'a str> for MethodName<'a> {
    type Error = MethodResolveError;
}
```

to write it once. But I think just removing this to  get `MethodNotFoundError` in `check_method` would be fine? We don't seem to charge any fees in that case.

Reply:
But this will change protocol, unfortunately, we must keep returning `MethodEmptyName `.

Reply:
Hm, are we actually sensitive to the *specific* error kind, and not just the fact that the error occurred? If we are, that's scary: in https://github.com/matklad/nearcore/blob/3f47dcd3d7c84195272cd67f426e8061260e2319/runtime/near-vm-errors/src/lib.rs#L37-L43

we more-or-less dump wasmer's internal errors to strings, and I bet their representation will change in the future. I guess, @bowenwang1996 is the best person to ask about that?

Reply:
Yes, we are sensitive to the exact error kind, that's why every VM change shall be done very careful.

Reply:
Hmm my understand from previously talked to @evgenykuzyakov  is "whether error happens or not" is merklelized in trie, not "what kind of error happened", so this is not a protocol change.

Reply:
The exact type of error does not matter, but it matters whether it is an error or not, so we cannot just remove this block.

Reply:
> so we cannot just remove this block.

The error would happen down the line, where we won't be able to find a method with an empty name in the list of methods exported by module. 

I guess the high-order bit here that we can refactor this, as long as we preserve the fact of error, and None vm outcome. We'll look into this after the PR is merged.